grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcTranscodingPathParser.java

```diff
@@ -0,0 +1,441 @@
+                return segment;
+            }
+            case '*': {
+                final PathSegment segment;
+                if (context.peek() == '*') {
+                    context.read();
+                    segment = new DeepWildcardPathSegment(context.nextPathVarIndex());
+                } else {
+                    segment = new WildcardPathSegment(context.nextPathVarIndex(), null);
+                }
+                checkArgument(!context.hasNext() || delimiters.contains(context.peek()),
+                              "Invalid path: context=%s", context);
+                return segment;
+            }
+            default: {
+                final StringBuilder literalBuilder = new StringBuilder().append(firstCh);
+                while (context.hasNext()) {
+                    final char c = context.peek();
+                    if (delimiters.contains(c)) {
+                        break;
+                    }
+                    literalBuilder.append(context.read());
+                }
+                return new LiteralPathSegment(literalBuilder.toString());
+            }
+        }
+    }
+
+    /**
+     * Parses 'Variable' part.
+     *
+     * <p>
+     * Variable = "{" FieldPath [ "=" Segments ] "}"
+     * </p>
+     */
+    private static PathSegment parseVariable(Context context, Delimiters delimiters) {
+        final StringBuilder fieldPathBuilder = new StringBuilder();
+        char c = 0;
+        while (context.hasNext()) {
+            c = context.peek();
+            if (delimiters.contains(c) || c == '=') {
+                break;
+            }
+            fieldPathBuilder.append(context.read());
+        }
+
+        final String fieldPath = fieldPathBuilder.toString();
+        checkArgument(!fieldPath.isEmpty(), "Invalid path: context=%s", context);
+
+        if (c == '=') {
+            // Consume '='.
+            context.read();
+
+            final List<PathSegment> segments = parseSegments(context, delimiters);
+
+            // Replace a single WildcardPathSegment with a new one which has a field name
+            // as its identifier in order to make the path variable human-readable.
+            if (segments.size() == 1 && segments.get(0) instanceof WildcardPathSegment) {
+                return new VariablePathSegment(
+                        fieldPath, ImmutableList.of(((WildcardPathSegment) segments.get(0))
+                                                            .withParentFieldPath(fieldPath)));
+            } else {
+                return new VariablePathSegment(fieldPath, segments);
+            }
+        } else {
+            // Treat '{name}' as '{name=*}' to specify a path variable index.
+            return new VariablePathSegment(fieldPath,
+                                           ImmutableList.of(new WildcardPathSegment(context.nextPathVarIndex(),
+                                                                                    fieldPath)));
```

Comment:
Good suggestion. But IMHO it won't help to speed up `populatePathVariable`  because `VariablePathSegment`s is converted into `PathVariable`s while building the transcoding service.